src/main/java/com/dell/isg/smi/service/device/discovery/manager/threads/DeviceIdentificationThread.java

```diff
@@ -14,107 +17,121 @@
-                discoverDeviceInfo.setDeviceType(discoveryDeviceType.getDiscoveryDeviceName());
-                discoverDeviceInfo.setStatus(DiscoveryDeviceStatus.DEVICE_IDENTFIED.name());
-                isDeviceIdentified = true;
-            }
-        }
-        return isDeviceIdentified;
-
-    }
-
-
-    private boolean identify(String ipAddress, DiscoveryDeviceProtocolEnum protocol, DiscoveryDeviceType discoveryDeviceType) {
-        boolean isDeviceIdentified = false;
-        switch (protocol) {
-        case HTTP:
-        case HTTPS:
-            isDeviceIdentified = DiscoverDeviceTypeUtil.processHttps(ipAddress, discoveryDeviceType);
-            break;
-        case TCP:
-            isDeviceIdentified = DiscoverDeviceTypeUtil.processTcp(ipAddress, discoveryDeviceType);
-            break;
-        case SSH:
-            isDeviceIdentified = DiscoverDeviceTypeUtil.processSsh(ipAddress, discoveryDeviceType);
-            break;
-        case TFTP:
-            isDeviceIdentified = DiscoverDeviceTypeUtil.processTftp(ipAddress, discoveryDeviceType);
-            break;
-        default:
-            isDeviceIdentified = false;
-        }
-        return isDeviceIdentified;
-    }
+	private DiscoveredDeviceInfo discoverDeviceInfo;
+	private DiscoveryDeviceConfigProvider discoveryDeviceConfigProvider;
+	private DiscoveryDeviceGroupEnum discoveryDeviceGroupEnum;
+
+	private static final Logger logger = LoggerFactory.getLogger(DeviceIdentificationThread.class.getName());
+
+	public DeviceIdentificationThread(DiscoveredDeviceInfo discoverDeviceInfo,
+			DiscoveryDeviceConfigProvider configProvider, DiscoveryDeviceGroupEnum discoveryDeviceGroupEnum) {
+		this.discoverDeviceInfo = discoverDeviceInfo;
+		this.discoveryDeviceConfigProvider = configProvider;
+		this.discoveryDeviceGroupEnum = discoveryDeviceGroupEnum;
+	}
+
+	@Override
+	public void run() {
+		try {
+			processCommand();
+		} catch (Exception e) {
+			logger.error(" Device idntification failed :" + e.getMessage());
+		}
+	}
+
+	private void processCommand() throws Exception {
+		String ipAddress = discoverDeviceInfo.getIpAddress();
+		logger.trace("Started device identification for IP : " + ipAddress);
+		if (IPRangeValidatorUtil.ping(ipAddress)) {
+			discoverDeviceInfo.setMacAddress(
+					IPRangeValidatorUtil.getMacAddressForIp(ipAddress, discoveryDeviceConfigProvider.getArpCommand()));
+			identifyDeviceByIP(ipAddress);
+		} else {
+			discoverDeviceInfo.setDeviceType(DiscoveryDeviceTypeEnum.UNKNOWN.value());
+			discoverDeviceInfo.setStatus(DiscoveryDeviceStatus.NO_DEVICE.name());
+		}
+	}
+
+	private void identifyDeviceByIP(String ipAddress) {
+		if (discoveryDeviceGroupEnum != null
+				&& EnumUtils.isValidEnum(DiscoveryDeviceGroupEnum.class, discoveryDeviceGroupEnum.value())) {
+			amI(ipAddress, discoveryDeviceGroupEnum);
+		} else if (amI(ipAddress, DiscoveryDeviceGroupEnum.SERVER)) {
```

Comment:
amI() is making IO calls and at higher risk of throwing exception, if does then you need to handle this case update the device with respective Type and Status